phases/07-transformers-deep-dive/12-kv-cache-flash-attention/code/main.py

```
@@ -117,12 +117,12 @@ def main():
 
     print(f"=== naive vs KV-cached decoding on N={N} tokens ===")
     print(f"naive attention ops:  {naive_ops}  (O(N^2) = {N * (N + 1) // 2})")
-    print(f"cached attention ops: {cached_ops}  (O(N) with per-step cost, unchanged)")
+    print(f"cached attention ops: {cached_ops}  (O(N) = {N})")
     print("outputs match (max abs diff over all tokens):",
           f"{max(abs(a - b) for va, vb in zip(naive, cached) for a, b in zip(va, vb)):.2e}")
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)